evaluate/task/test-integration/task_test.go

```diff
@@ -22,22 +21,14 @@ func TestTaskWriteTestsRun(t *testing.T) {
 
 	validate := func(t *testing.T, tc *tasktesting.TestCaseTask) {
 		t.Run(tc.Name, func(t *testing.T) {
-			resultPath := t.TempDir()
-
-			logOutput, logger := log.Buffer()
-			defer func() {
-				if t.Failed() {
-					t.Logf("Logging output: %s", logOutput.String())
-				}
-			}()
-			repository, cleanup, err := evaluatetask.TemporaryRepository(logger, tc.TestDataPath, tc.RepositoryPath)
-			assert.NoError(t, err)
-			defer cleanup()
-
-			taskWriteTests, err := evaluatetask.TaskForIdentifier(evaluatetask.IdentifierWriteTests)
-			require.NoError(t, err)
-
-			tc.Validate(t, taskWriteTests, repository, resultPath, logger)
+			tc.Validate(t,
+				func(logger *log.Logger, testDataPath string, repositoryPathRelative string) (repository evaltask.Repository, cleanup func(), err error) {
+					return evaluatetask.TemporaryRepository(logger, testDataPath, repositoryPathRelative)
+				},
+				func() (task evaltask.Task, err error) {
+					return evaluatetask.TaskForIdentifier(evaluatetask.IdentifierWriteTests)
+				},
+			)
```

Comment:
I guess we still cannot move this entirely into the `testing`  package cause we would have a cyclic import between testing and implementation :disappointed_relieved:

Reply:
but we also cannot move them into one cause we want the `testing` package to use it in the `integration` tests... grrrrrr. At this point... why not just have it in one single package? It all tests the same thing? Forcing it to be split is just super annoying at this point.

Reply:
I could get rid of the second callback. the first one is still there. We need to keep refactoring to get rid of that.

Reply:
Would just chuck everything into one package cause it is all testing the task... But it's already better than before.

Reply:
> Would just chuck everything into one package cause it is all testing the task... But it's already better than before.

That would just hide the fact that our non-testing package structure is still not perfect. But we are getting there.